### PR TITLE
Added quick actions for qr code scanner and certificates

### PR DESCRIPTION
**Pull request by ln-12**
## Description
This PR is meant to complete https://github.com/corona-warn-app/cwa-app-android/pull/4685, so that both Android and iOS can get all four app shortcuts/quick actions. I followed the suggestions in https://github.com/corona-warn-app/cwa-app-android/pull/4642 to hopefully meet all requirements. If not, please let me know.

For now I only added german and english localization, the other languages need to be done.

## Link to Jira
[https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5414](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5414)

## Screenshots
7A34CE80-6049-4C8F-9790-F191F0302B00

### CODE REVIEW

**Review by CV113 — APPROVED** [review]
UA approved

**Review by marcussc — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift [hunk 1]
```diff
@@ -50,13 +52,17 @@ enum QuickAction: String {
 			return
 		}
 
-		var shortcutItems = [UIApplicationShortcutItem(type: QuickAction.diaryNewEntry.rawValue, localizedTitle: AppStrings.QuickActions.contactDiaryNewEntry, localizedSubtitle: nil, icon: UIApplicationShortcutIcon(templateImageName: "book.closed"))]
+		var shortcutItems = [
+			UIApplicationShortcutItem(type: QuickAction.diaryNewEntry.rawValue, localizedTitle: AppStrings.QuickActions.contactDiaryNewEntry, localizedSubtitle: nil, icon: UIApplicationShortcutIcon(templateImageName: "book.closed")),
+			UIApplicationShortcutItem(type: QuickAction.eventCheckin.rawValue, localizedTitle: AppStrings.QuickActions.eventCheckin, localizedSubtitle: nil, icon: UIApplicationShortcutIcon(templateImageName: "Icons_Tabbar_Checkin")),
+			UIApplicationShortcutItem(type: QuickAction.showCertificates.rawValue, localizedTitle: AppStrings.QuickActions.showCertificates, localizedSubtitle: nil, icon: UIApplicationShortcutIcon(templateImageName: "Icons_Tabbar_Certificates"))
+		]
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift:59]:
```suggestion
		var shortcutItems = [
			UIApplicationShortcutItem(type: QuickAction.diaryNewEntry.rawValue, localizedTitle: AppStrings.QuickActions.contactDiaryNewEntry, localizedSubtitle: nil, icon: UIApplicationShortcutIcon(templateImageName: "book.closed")),
			UIApplicationShortcutItem(type: QuickAction.showCertificates.rawValue, localizedTitle: AppStrings.QuickActions.showCertificates, localizedSubtitle: nil, icon: UIApplicationShortcutIcon(templateImageName: "Icons_Tabbar_Certificates"))
		]
```

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift [hunk 2]
```diff
@@ -50,13 +52,17 @@ enum QuickAction: String {
...
 		
 		let status = AVCaptureDevice.authorizationStatus(for: .video)
 		if status == .authorized || status == .notDetermined {
-			// dont show event checkin action if no camera access granted
+			// dont show camera related actions if no camera access is granted
 			shortcutItems.append(
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift:64]:
```suggestion
			shortcutItems.append(
				contentsOf: [
					UIApplicationShortcutItem(type: QuickAction.qrCodeScanner.rawValue, localizedTitle: AppStrings.QuickActions.qrCodeScanner, localizedSubtitle: nil, icon: UIApplicationShortcutIcon(templateImageName: "qrcode.viewfinder")),
					UIApplicationShortcutItem(type: QuickAction.eventCheckin.rawValue, localizedTitle: AppStrings.QuickActions.eventCheckin, localizedSubtitle: nil, icon: UIApplicationShortcutIcon(templateImageName: "Icons_Tabbar_Checkin"))
				]
			)
```

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift [hunk 3]
```diff
@@ -50,13 +52,17 @@ enum QuickAction: String {
...
-				UIApplicationShortcutItem(type: QuickAction.eventCheckin.rawValue, localizedTitle: AppStrings.QuickActions.eventCheckin, localizedSubtitle: nil, icon: UIApplicationShortcutIcon(templateImageName: "qrcode.viewfinder"))
+				UIApplicationShortcutItem(type: QuickAction.qrCodeScanner.rawValue, localizedTitle: AppStrings.QuickActions.qrCodeScanner, localizedSubtitle: nil, icon: UIApplicationShortcutIcon(templateImageName: "qrcode.viewfinder"))
 			)
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift:66]:
```suggestion

```

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift [hunk 4]
```diff
@@ -104,6 +110,33 @@ extension AppDelegate {
...
+			coordinator.checkinTabCoordinator?.viewController.presentedViewController?.dismiss(animated: false, completion: nil)
+			
+		case QuickAction.eventCheckin.rawValue:
+			Log.info("Shortcut: Event checkin 📷", log: .ui)
+			guard let tabBarController = coordinator.tabBarController else { return }
+			tabBarController.selectedIndex = 3
+			
+			// dismiss an overlaying, modally presented view controller
+			coordinator.checkinTabCoordinator?.viewController.presentedViewController?.dismiss(animated: false, completion: nil)
+
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift:139]:
```suggestion
						// open qr code scanner for fast event checkin
			coordinator.checkinTabCoordinator?.showQRCodeScanner()

```

**Review by 30mar — CHANGES_REQUESTED** [review]
@ln-12  thanks alot for you effort, I tested the project locally and I've added some comments, feel free to reply to them if you need to clarify something. :)

File: src/xcode/ENA/ENA/Resources/Localization/en.lproj/Localizable.strings [hunk 5]
```diff
@@ -2298,10 +2298,15 @@
 "Contact_Journal_Notes_Description" = "You can enter notes here that will help you to better estimate your risk of infection.\n\nWrite down additional circumstances of your encounter or stay here that could have an effect on your risk of infection, for example, how close you were to other people and what you did (examples: “sat close together”, “played sports”, “little space”, “we sang”).\n\nThese notes can help you refresh your memory later. If necessary, they will also let you warn the people you have been with at an early stage, as well as help the public health authorities trace potential chains of infection.";
 
 /* Quick Actions a.k.a. shortcuts */
-"QuickAction_newContactDiaryEntry" = "Add Journal Entry for Today";
+
```
[the remark this answers is not in the document]
Comment (ln-12) [reply src/xcode/ENA/ENA/Resources/Localization/en.lproj/Localizable.strings:2301]: Done

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (ln-12) [reply src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift]: Hey, thanks for the suggestions. I added all of these.

**Comment by ln-12** [comment]
@marcussc That because of @30mar's comment above: https://github.com/corona-warn-app/cwa-app-ios/pull/4238#discussion_r817792932

Of course, I can remove it again, but you have to have the same opinion on what's needed...

**Comment by marcussc** [comment]
@ln-12 @30mar so we had a short internal discussion and decided it makes more sense to not show the QR Code scanner on the "Check-ins anzeigen" Quickaction. It used to do this before we had the universal QRCodescanner. 
Sorry for the confusion that my have caused.

**Comment by ln-12** [comment]
@marcussc Nevermind, I changed it back :)

**Comment by marcussc** [comment]
@ln-12 thank you. One last small thing, please move the "Check-ins anzeigen" to the third position. Then I don't see anything that stands in the way of merging your PR :)

**Comment by ln-12** [comment]
Alright, I reordered the actions.

**Review by 30mar — APPROVED** [review]

**Review by marcussc — APPROVED** [review]

**Pull request merged**